### PR TITLE
[tree] Prefer Long64_t to int64_t for the number of TTree entries 

### PR DESCRIPTION
**Pull request by eguiraud**
For consistency with the rest of libTree.

### CODE REVIEW

File: tree/tree/inc/ROOT/RFriendInfo.hxx
Comment (vepadulano) [inline tree/tree/inc/ROOT/RFriendInfo.hxx:22]: Is it better to use `Rtypes` or `RTypescore` in this case?

File: tree/tree/inc/ROOT/RFriendInfo.hxx
Comment (vepadulano) [inline tree/tree/inc/ROOT/RFriendInfo.hxx:84]: Here and in similar places, maybe we could have 
```suggestion
                  Long64_t nEntries = TTree::kMaxEntries, TVirtualIndex *indexInfo = nullptr);
```
instead?

**Review by vepadulano — APPROVED** [review]
LGTM overall! Minor comments follow

File: tree/tree/inc/ROOT/RFriendInfo.hxx
Comment (eguiraud) [reply tree/tree/inc/ROOT/RFriendInfo.hxx:22]: $ rg 'Rtypes\.h' tree/dataframe | wc -l
11
$ rg 'Rtypescore\.h' tree/dataframe | wc -l
0

:man_shrugging:

File: tree/tree/inc/ROOT/RFriendInfo.hxx
Comment (eguiraud) [reply tree/tree/inc/ROOT/RFriendInfo.hxx:84]: would need to check the code for places where we match this value, e.g. in TTreeProcessorMT there is a `fGlobalRange.second != std::numeric_limits<Long64_t>::max()`. and I don't really understand https://github.com/root-project/root/pull/12352 so I'm not sure we'd incur in it here (@hahnjo ?).

File: tree/tree/inc/ROOT/RFriendInfo.hxx
Comment (vepadulano) [reply tree/tree/inc/ROOT/RFriendInfo.hxx:22]: Yeah the spelling is not the most brilliant here :sweat_smile: 

```
$: grep -r 'RtypesCore.h' tree/dataframe | wc -l
23
```

So maybe in a separate PR we should decide on which of the two headers is better to use everywhere? There must be some difference that makes one useful in some cases where the other isn't.

Comment (pcanal) [reply tree/tree/inc/ROOT/RFriendInfo.hxx:22]: > must be some difference that makes one useful in some cases where the other isn't.

One is older than the other.  One (the same one) include the others.  If `RtypesCore.h` is sufficient, it should be preferred.  Essentially `Rtypes.h` is `RtypesCore.h` + `ClassDef` declarations.

File: tree/tree/inc/ROOT/RFriendInfo.hxx
Comment (pcanal) [reply tree/tree/inc/ROOT/RFriendInfo.hxx:84]: `static constexpr Long64_t kMaxEntries = std::numeric_limits<Long64_t>::max();`

So it really is a semantic choice.

Comment (eguiraud) [reply tree/tree/inc/ROOT/RFriendInfo.hxx:84]: that doesn't really change the fact that we want to be consistent in which one we use

Comment (pcanal) [reply tree/tree/inc/ROOT/RFriendInfo.hxx:84]: If your number is a number of entries, I recommend using `TTree::kMaxEntries`

**Comment by guitargeek** [comment]
@eguiraud, what's the status of this PR? Can we just merge it?

**Comment by eguiraud** [comment]
@guitargeek besides the dumb conflict to resolve, this is probably ok to merge as long as people (namely @vepadulano and @pcanal) are ok with keeping ` std::numeric_limits<Long64_t>::max()` instead of `kMaxEntries` sometimes.

**Comment by pcanal** [comment]
I resolved the merge conflict and updated to use `kMaxEntries` (since this is the semantic accurate alias).

**Review by pcanal — APPROVED** [review]

**Pull request merged**
